cmpserver/plugin/plugin.go

```diff
@@ -286,18 +286,19 @@ func (s *Service) matchRepositoryGeneric(stream MatchRepositoryStream) error {
 	return nil
 }
 
-func (s *Service) matchRepository(ctx context.Context, workdir string, envEntries []*apiclient.EnvEntry) (bool, error) {
+func (s *Service) matchRepository(ctx context.Context, workdir string, envEntries []*apiclient.EnvEntry) (bool, bool, error) {
 	config := s.initConstants.PluginConfig
+
 	if config.Spec.Discover.FileName != "" {
```

Comment:
Do we now need a nil check before we start accessing `config.Spec.Discover` fields?

Reply:
No those are concrete structs and not pointers so nil check is not needed.